src/runtime_src/driver/xclng/xrt/user_common/sensor.h

```diff
@@ -0,0 +1,12 @@
+#ifndef SENSOR_H
+#define SENSOR_H
+
+#include <boost/property_tree/ptree.hpp>
+#include <boost/property_tree/json_parser.hpp>
+#include <boost/foreach.hpp>
+
+static boost::property_tree::ptree gSensorTree;
```

Comment:
No static variables in header files.  If the header is include multiple times, each compilation unit will have an instance of the static variable.  Move to sensor.cpp

Reply:
@stsoe 
Please review my latest changes. I made this a singleton class where the ptree object is a private member variable. I've written wrappers around ptree::get and ptree::put.

Reply:
I am confused, where is the global used?  Who calls Instance()?

Reply:
The property tree (previously gSensorTree) is no-longer global. Instance() provides a reference to the singleton class where sensorTree::put() and sensorTree::get() are wrappers around boost::property_tree::ptree::put() and ::get().

Reply:
I still don't see where Instance() is called?

Reply:
In xbutil.h (See line 239) wherever sensorTree::put() is called it must be called like so:


```
sensorTree &tree = sensorTree::Instance();
tree.put( val );
```

Reply:
ok. thanks.